processor.go

```diff
@@ -145,6 +148,77 @@ func (c *OAuthProcessorConfig) Processor(params map[string]string) (RequestProce
 	}, nil
 }
 
+type Sigv4ProcessorConfig struct {
+	AccessKey string `json:"access_key"`
+	SecretKey string `json:"secret_key"`
+}
+
+var _ ProcessorConfig = (*Sigv4ProcessorConfig)(nil)
+
+func (c *Sigv4ProcessorConfig) Processor(params map[string]string) (RequestProcessor, error) {
+
+	if len(c.AccessKey) == 0 {
+		return nil, errors.New("missing access key")
+	}
+	if len(c.SecretKey) == 0 {
+		return nil, errors.New("missing secret key")
+	}
+
+	return func(r *http.Request) error {
+		// NOTE: Sigv4 has defenses against request forgery and reuse.
+		// This does *not* make those guarantees, and likely can not.
+
+		var (
+			service string
+			region  string
+			date    time.Time
+			err     error
+		)
+
+		// Parse the auth header to get the service, region, and date
+		authHeader := r.Header.Get("Authorization")
+		if authHeader == "" {
+			return errors.New("expected request to contain an Authorization header")
+		}
+
+		for _, section := range strings.Split(authHeader, " ") {
+			section = strings.TrimRight(section, ",")
+			keyValuePair := strings.SplitN(section, "=", 2)
+			if len(keyValuePair) != 2 {
+				continue
+			}
+
+			if keyValuePair[0] == "Credential" {
+				credParts := strings.Split(keyValuePair[1], "/")
+				if len(credParts) != 5 {
+					return errors.New("invalid credential in auth header")
+				}
+
+				dateStr := credParts[1]
+				date, err = time.Parse("20060102T150405Z", dateStr)
+				if err != nil {
+					return err
+				}
+
+				service = credParts[2]
+				region = credParts[3]
+				break
+			}
+		}
+
```

Comment:
Should we check here that we found a good credential in the header or is it okay for `service`, `region`, and `date` to be empty?

Reply:
great catch, I missed the error handling in the refactor 😅